docs/rst/tutorials/pkcs11_tutorial.rst

```diff
@@ -0,0 +1,135 @@
+.. _vulcanexus_pkcs11_tutorial:
+
+ROS 2 Security improved with Hardware Security Modules (HSM) using Vulcanexus Core
+==========================================================================================
+
+.. contents:: Table of Contents
+    :depth: 2
+    :local:
+    :backlinks: none
+
+Background
+----------
+
+`ROS 2 Security <https://docs.ros.org/en/humble/Concepts/About-Security.html>`_ is implemented by means of the `DDS Security specification <https://www.omg.org/spec/DDS-SECURITY/About-DDS-SECURITY/>`_.
+The security files required by the aforementioned specification are encapsulated within the ROS 2 security enclaves.
+This approach allows to set different security policies for different nodes, processes, users, or devices by defining different enclaves with the corresponding permissions.
+However, ROS 2 currently requires that the private key used to encrypt and decrypt the enclave information is also stored within the enclave, which might be a security vulnerability.
+Hardware Security Modules (HSM) can be used in order to mitigate this vulnerability and PKCS #11 standard provides an API to interact with these devices.
+Vulcanexus Core provides the required support for HSM devices so the private key, instead of being stored in the filesystem, it could be inside the HSM device.
+
+This tutorial provides step-by-step instructions to improve ROS 2 Security of a talker/listener demo with HSM using Vulcanexus Core.
+
+Prerequisites
+-------------
+
+This tutorial does not require a physical HSM device connected to the test environment.
+Instead, a software HSM emulator is going to be installed and used to run this tutorial.
+
+Run the following commands to install the required packages:
+
+.. code-block:: bash
+
+    sudo apt update && apt install --yes --no-install-recommends \
+      softhsm2 \
+      gnutls-bin
```

Comment:
```suggestion
```suggestion
Instead, a software HSM emulator is going to be installed and used.
Run the following commands to install the required packages (administrative privileges may be required):

.. code-block:: bash
    apt update
    apt install --yes --no-install-recommends \
      softhsm2 \
      gnutls-bin
```
```